ReliefApplications/ems-frontend
# AB#107541 - Email Notifications ES651 Distribution change to incorporate common services

# Description
Added option to select users emails from common services in step 3 (Distribution List step)

## Useful links

ES651- https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/107541

Backend Related Ticket - https://github.com/ReliefApplications/ems-backend/pull/1174

## Type of change



# How Has This Been Tested?

Old behavior:
- When user navigate to the DL page, there are 3 email input options, namely Add manually, select with filter, use combination
- Add manually: user will manually type in the required email address.
- Select with Filter: Here user will be presented with a data type field which allows the user to either select a resource or reference data. Based on the selection, respective forms would appear. User can then continue with the field selection, filter logic and display the data accordingly.
- Use combination: This would be a combination of add manually and select with filter feature

New behavior
- When user navigate to the DL page, there will be 4 email input options, namely Add manually, select with filter, Select from Users table (name subjected to change) and use combination
- Add manually: user will manually type in the required email address.
- Select with Filter: Here user will be presented with Resource dropdown which contains the list of all the resources. After the form selection, user can then continue with the field selection, filter logic and display the data accordingly.
- Select from Common Services: 
- Use combination: This would be a combination of add manually, select with filter, Select from Users table feature in accordion fashion

- [ ] EN creation flow
- [ ] EN edit flow
- [ ] EN direct send then navigate to step 3

# Checklist:

( * == Mandatory ) 


# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145
